library/core/class.request.php

```diff
@@ -1415,16 +1415,17 @@ public function _unsetRequestArguments($paramsType) {
     public function url($path = '', $withDomain = false, $ssl = null) {
         static $allowSSL = null;
         if ($allowSSL === null) {
-            $allowSSL = c('Garden.AllowSSL', false);
+            $allowSSL = c('Garden.AllowSSL', c('Garden.ForceSSL'));
         }
+
         static $rewrite = null;
         if ($rewrite === null) {
             $rewrite = val('X_REWRITE', $_SERVER, c('Garden.RewriteUrls', true));
         }
 
         if (!$allowSSL) {
             $ssl = null;
-        } elseif ($withDomain === 'https') {
+        } elseif ($withDomain === 'https' || c('Garden.ForceSSL')) {
```

Comment:
@linc @tburry Is this a good idea? Without this, I still get an HTTP canonical link when I access a HTTP page with ForceSSL to true.